app/src/main/java/com/android/sample/ui/createRecipe/CategoryScreen.kt

```diff
@@ -0,0 +1,171 @@
+import com.android.sample.resources.C.Dimension.PADDING_32
+import com.android.sample.resources.C.Dimension.PADDING_8
+import com.android.sample.resources.C.Tag.INITIAL_RECIPE_STEP
+import com.android.sample.resources.C.TestTag.Category.BUTTON_TEST_TAG
+import com.android.sample.resources.C.TestTag.Category.DROPDOWN_CORNER_RADIUS
+import com.android.sample.resources.C.TestTag.Category.DROPDOWN_TEST_TAG
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.theme.Typography
+import com.android.sample.ui.utils.PlateSwipeButton
+import com.android.sample.ui.utils.PlateSwipeScaffold
+
+@Composable
+fun CategoryScreen(
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel,
+) {
+  PlateSwipeScaffold(
+      navigationActions = navigationActions,
+      selectedItem = Route.CREATE_RECIPE,
+      showBackArrow = true,
+      content = { paddingValues ->
+        CategoryContent(
+            currentStep = INITIAL_RECIPE_STEP,
+            navigationActions = navigationActions,
+            createRecipeViewModel = createRecipeViewModel,
+            modifier = Modifier.fillMaxSize().padding(paddingValues))
+      })
+}
+
+/**
+ * Composable function to optionally select a category for a recipe.
+ *
+ * @param modifier Modifier to be applied to the screen.
+ * @param currentStep The current step in the recipe creation process.
+ * @param navigationActions Actions for navigating between screens.
+ * @param createRecipeViewModel ViewModel for managing the recipe creation process.
+ */
+@Composable
+fun CategoryContent(
+    modifier: Modifier = Modifier,
+    currentStep: Int,
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel
+) {
+  val noCategoryString = stringResource(R.string.no_category)
+  val categories = listOf(noCategoryString) + Recipe.getCategories()
+  val selectedCategory = remember { mutableStateOf(createRecipeViewModel.getRecipeCategory()) }
+  val expanded = remember { mutableStateOf(false) }
+
+  Box(modifier = modifier.padding(PADDING_8.dp), contentAlignment = Alignment.TopCenter) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Top) {
+          // Display the progress bar for the current step
+          RecipeProgressBar(currentStep = currentStep)
+
+          Spacer(modifier = Modifier.weight(1f))
+
+          // Title
+          Text(
+              text = stringResource(R.string.select_category),
+              style = Typography.displayLarge,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(horizontal = PADDING_16.dp)
+                      .testTag("CategoryTitle"),
```

Comment:
Please define the fraction in the `TestTag` object within the constants file (`resources/C.kt`).